sklearn/neighbors/tests/test_nca.py

```diff
@@ -531,18 +531,19 @@ def test_parameters_valid_types(param, value):
     nca.fit(X, y)
 
 
-def test_nca_feature_names_out():
+@pytest.mark.parametrize("n_components", range(1, iris.data.shape[1]))
+def test_nca_feature_names_out(n_components):
```

Comment:
Looking at this test, it can serve as a non-regression test when `n_compenents is not None`. We don't need to add an additional test.